lib/ansible/modules/network/infinity/infinity.py

```diff
@@ -0,0 +1,650 @@
+            network_location=-1):
+        """
+        add  new nework into    Infinity via rest api  into given supernet or default supernet
+        required fields=['network_name', 'network_family', 'network_type',  'network_address','network_size' ]
+
+        """
+        method = 'post'
+        resource_url = 'networks'
+        response = None
+        if network_name is None or network_address is None or network_size is None:
+            self.module.exit_json(
+                msg="You must specify  those the  options 'network_name', 'network_address' and 'network_size'")
+
+        if not network_family:
+            network_family = '4'
+        if not network_type:
+            network_type = 'lan'
+        if not network_location:
+            network_location = -1
+        payload_data = {
+            "network_name": network_name,
+            'network_address': network_address,
+            'network_size': network_size,
+            'network_family': network_family,
+            'network_type': network_type,
+            'network_location': network_location}
+        response = self._get_api_call_ansible_handler(
+            method='post',
+            resource_url=resource_url,
+            stat_codes=[200],
+            payload_data=payload_data)
+        return response
+
+
+def main():
+
+    fields = {
+        "server": {
+            "required": True,
+            "type": "str"},
+        "username": {
+            "required": True,
+            "type": "str"},
+        "password": {
+            "required": True,
+            "type": "str"},
+        "network_id": {
+            "required": False},
+        "ip_address": {
+            "required": False,
+            "type": "str"},
+        "network_name": {
+            "required": False,
+            "type": "str"},
+        "network_location": {
+            "required": False,
+            "type": "int",
+            "default": -1},
+        "network_family": {
+            "required": False,
+            "type": "str",
+            "choices": [
+                '4',
+                '6',
+                'dual']},
+        "network_type": {
+            "required": False,
+            "choices": [
+                'lan',
+                'shared_lan',
+                'supernet']},
+        "network_address": {
+            "required": False,
+            "type": "str"},
+        "network_size": {
+            "required": False,
+            "type": "str"},
+        "action": {
+            "required": True,
+            "choices": [
+                'get_network',  # works
+                'reserve_next_available_ip',  # works
+                'release_ip',  # work
+                'delete_network',  # work
+                'reserve_network',  # work
+                'release_network',  # work
+                'add_network',
+                'get_network_id'],  # work
+        },
+    }
+
+    # mutually_exclusive = [ ["network_id", "network_name"] ]
```

Comment:
Is this needed?

If they are then they should be added to the `DOCUMENTATION`

For usage see https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec

http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block